core/arch/arm/include/kernel/early_ta.h

```diff
@@ -19,5 +21,11 @@ struct early_ta {
 
 #define __early_ta __section(".rodata.early_ta" __SECTION_FLAGS_RODATA)
 
+#define for_each_early_ta(_ta) \
+	for (_ta = &__rodata_early_ta_start; _ta < &__rodata_early_ta_end; \
+	     _ta = (const struct early_ta *)				   \
+		   ROUNDUP((vaddr_t)_ta + sizeof(*_ta) + _ta->size,	   \
+			   __alignof__(struct early_ta)))
```

Comment:
maybe:
```c
	for (_ta = &__rodata_early_ta_start; _ta < &__rodata_early_ta_end; \
	     _ta = (__typeof__(_ta))ROUNDUP((vaddr_t)_ta + sizeof(*_ta) +  \
					    _ta->size, __alignof__(*_ta)))
```

Reply:
I would argue this way is slightly less clear because we lose the "struct early_ta" so when we want to use this macro we don't know immediately the type of the variable to declare. So having the type here is a bit like a function prototype if you wish.

In any case I prefer not to change the code in a commit with title "move for_each_early_ta() macro...".

Reply:
Matter of taste maybe. Note `&__rodata_early_ta_start` refers to a `const struct early_ta *` so compiler should complain if `_ta` is not of that type, aside for the `const` attribute maybe.

Reply:
If it should be `__typeof__()` anything it should be of `&__rodata_early_ta_start`, but I prefer it as it is as explained by @jforissier.